src/subsai/models/whisper_api_model.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Whisper API Model
+
+See [openai/whisper](https://platform.openai.com/docs/guides/speech-to-text)
+"""
+
+import os
+import ffmpeg
+from subsai.models.abstract_model import AbstractModel
+from subsai.utils import _load_config
+from openai import OpenAI
+from pysubs2 import SSAFile
+
+def convert_video_to_audio_ffmpeg(video_file, output_ext="mp3"):
+    # Construct the output file name
+    filename, ext = os.path.splitext(video_file)
+    output_file = f"{filename}.{output_ext}"
+    
+    # Execute the ffmpeg conversion
+    (
+        ffmpeg
+        .input(video_file)
+        .output(output_file)
+        .overwrite_output()
+        .run(quiet=True)
+    )
+
+    return output_file
+
+
+class WhisperAPIModel(AbstractModel):
+    model_name = 'openai/whisper'
+    config_schema = {
+            # load model config
+            'model_type': {
+                'type': list,
+                'description': "One of the official model names listed by `whisper.available_models()`, or "
+                               "path to a model checkpoint containing the model dimensions and the model "
+                               "state_dict.",
+                'options': ['whisper-1'],
+                'default': 'whisper-1'
+            },
+            'api_key': {
+                'type': str,
+                'description': "Your OpenAI API key",
+                'options': None,
+                'default': None
+            },
+        }
+
+    def __init__(self, model_config):
+        # config
+        self.model_type = _load_config('model_type', model_config, self.config_schema)
+        self.api_key = _load_config('api_key', model_config, self.config_schema)
+
+        self.client = OpenAI(api_key=self.api_key)
+        
+
+    def transcribe(self, media_file) -> str:
+        audio_file_path = convert_video_to_audio_ffmpeg(media_file)
+        audio_file = open(audio_file_path, "rb")
```

Comment:
Their docs state that
 > File uploads are currently limited to 25 MB

I believe you'll need to handle this test case before processing ?

Reply:
I added support for chunking so audio will be sent in batches to avoid triggering the limit.

I could think about adding some overlap between the chunks but I'm not sure of the ideal way to then recombine the subtitles, from some quick testing my current solution seems to work fine.

Reply:
Yes, It's not easy. 
The problem with this approach is you that may cut the audio mid-speech and so the transcription may not be that good around the edges.

A good way I think would be something similar to what I did here:

> * [VAD example](https://github.com/abdeladim-s/subsai/blob/main/examples/subsai_vad.ipynb): process long audio files using [silero-vad](https://github.com/snakers4/silero-vad). <a target="_blank" href="https://colab.research.google.com/github/abdeladim-s/subsai/blob/main/examples/subsai_vad.ipynb">
  <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
</a>

I used a VAD to cut long files into chunks only when there is no speech, transcribe the chunks one by one and then reconstruct the subtitles.

Reply:
I'd be interested in having a go at this but I would split this into a different issue and use the basic chunking code in this PR

Reply:
Okey sounds good. 
Thanks a lot.